lib/publiccloud/provider.pm

```diff
@@ -191,6 +191,38 @@ sub run_ipa {
 }
 
 
+=head2 create_instance
+
+Creates an instance on the public cloud provider using ipa command without cleanup
+
+=cut
+sub create_instance {
+    my ($self, %args) = @_;
+
+    return $self->ipa(
+        instance_type => get_var('PUBLIC_CLOUD_INSTANCE_TYPE'),
+        cleanup       => 0,
+        image_id      => $args{image}
+    )
+}
+
+=head2 run_ssh_command
+
+Runs a command C<cmd> via ssh in the given VM.
+C<instance> should be an object returned by ipa subroutine, e.g. $provider->ipa()
+
+=cut
+sub run_ssh_command {
+    my ($self, %args) = @_;
```

Comment:
function will fail in case at least one arg is missing , can you add `die` ? or generic one based `args`  size or per argument